internal/storage/overlay/overlay.go

```diff
@@ -53,6 +53,11 @@ func Mount(ctx context.Context, layerPaths []string, upperdirPath, workdirPath,
 
 	options := []string{"lowerdir=" + lowerdir}
 	if upperdirPath != "" {
+		if _, err := os.Stat(upperdirPath); err == nil {
+			// Upperdir already exists. This means we're likely using the sandbox containers
+			// scratch as the containers scratch so we need to make a new name for the upperdir.
```

Comment:
I feel like we don't really need to specifically call out that we're using the sanbox container's scratch here. This is pretty generic so this comment should just say:

```
Upperdir already exists. This means we are reusing a scratch space from another container. Create a new dir with a random name to use for this container's upperdir.
```